### ISSUE TITLE
The Accuracy Problem

### ISSUE THREAD

**Issue by mobulan** [uitml/noHub#2]
I followed the steps to reproduce your experiment, but the average accuracy obtained was much lower than shown. Please tell me where I went wrong or is there anything else I needs attention? The experimental settings, datasets, and accuracy are as follows.

### Datasets
```
C:.
├─test
│  ├─n01930112
│  ├─n01981276
│  ├─n02099601
│  ├─n02110063
│  ├─n02110341
│  ├─n02116738
│  ├─n02129165
│  ├─n02219486
│  ├─n02443484
│  ├─n02871525
│  ├─n03127925
│  ├─n03146219
│  ├─n03272010
│  ├─n03544143
│  ├─n03775546
│  ├─n04146614
│  ├─n04149813
│  ├─n04418357
│  ├─n04522168
│  └─n07613480
├─train
│  ├─n01532829
│  ├─n01558993
│  ├─n01704323
│  ├─n01749939
│  ├─n01770081
│  ├─n01843383
│  ├─n01910747
│  ├─n02074367
│  ├─n02089867
│  ├─n02091831
│  ├─n02101006
│  ├─n02105505
│  ├─n02108089
│  ├─n02108551
│  ├─n02108915
│  ├─n02111277
│  ├─n02113712
│  ├─n02120079
│  ├─n02165456
│  ├─n02457408
│  ├─n02606052
│  ├─n02687172
│  ├─n02747177
│  ├─n02795169
│  ├─n02823428
│  ├─n02966193
│  ├─n03017168
│  ├─n03047690
│  ├─n03062245
│  ├─n03207743
│  ├─n03220513
│  ├─n03337140
│  ├─n03347037
│  ├─n03400231
│  ├─n03476684
│  ├─n03527444
│  ├─n03676483
│  ├─n03838899
│  ├─n03854065
│  ├─n03888605
│  ├─n03908618
│  ├─n03924679
│  ├─n03998194
│  ├─n04067472
│  ├─n04243546
│  ├─n04251144
│  ├─n04258138
│  ├─n04275548
│  ├─n04296562
│  ├─n04389033
│  ├─n04435653
│  ├─n04443257
│  ├─n04509417
│  ├─n04515003
│  ├─n04596742
│  ├─n04604644
│  ├─n04612504
│  ├─n06794110
│  ├─n07584110
│  ├─n07697537
│  ├─n07747607
│  ├─n09246464
│  ├─n13054560
│  └─n13133613
└─val
    ├─n01855672
    ├─n02091244
    ├─n02114548
    ├─n02138441
    ├─n02174001
    ├─n02950826
    ├─n02971356
    ├─n02981792
    ├─n03075370
    ├─n03417042
    ├─n03535780
    ├─n03584254
    ├─n03770439
    ├─n03773504
    ├─n03980874
    └─n09256479
```
### Logs
https://github.com/mobulan/nohub/tree/main/run-20230910_193734-s5pjdtdl

### Acc
```
(20) PS C:\Experiment\Code\Github\noHub> python src/evaluate.py --embedding nohub --classifier siamese --dataset mini --arch resnet18 --checkpoint tim/mini/softmax/resnet18/model_best.pth.tar
2023-09-10 19:37:30,631 - INFO: Running evaluation with config:
config = None
dataset = mini
embedding = nohub
classifier = siamese
arch = resnet18
checkpoint = tim\mini\softmax\resnet18\model_best.pth.tar
batch_size = 128
n_workers = 4
use_cached = False
cache_dir = None
n_episodes = 10000
episode_batch_size = 500
n_shots = 1
n_ways = 5
n_queries = 15
eval_split = test
log_level = INFO
wandb_tags = None
nohub.
  init = pca
  out_dims = 400
  initial_dims = None
  kappa = 0.5
  perplexity = 45.0
  re_norm = True
  eps = 1e-12
  p_sim = vmf
  p_rel_tol = 0.01
  p_abs_tol = None
  p_betas = (None, None)
  pca_mode = base
  n_iter = 50
  learning_rate = 0.1
  loss_weights = (0.2,)
siamese.
  lam = 10
  alpha = 0.2
  epsilon = 0.001
  optimal_transport_max_iter = 1000
  n_iter = 20
compute_hubness = True
k_occ.
  k = 5
  metric = sqeuclidean

2023-09-10 19:37:30,631 - WARNING: Could not find EXPERIMENT_ID in environment variables. Using generated id '0wzgiim1'.
wandb: Currently logged in as: mbl. Use `wandb login --relogin` to force relogin
wandb: Tracking run with wandb version 0.15.10
wandb: Run data is saved locally in C:\Experiment\Code\Github\noHub\wandb\run-20230910_193734-s5pjdtdl
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run mini-resnet18-nohub-siamese-0wzgiim1
wandb:  View project at https://wandb.ai/mbl/uncategorized
wandb:  View run at https://wandb.ai/mbl/uncategorized/runs/s5pjdtdl
2023-09-10 19:37:39,416 - INFO: Loading weights from file: 'C:\Experiment\Code\Github\noHub\models\tim\mini\softmax\resnet18\model_best.pth.tar'
2023-09-10 19:37:40,279 - INFO: Extracting features.
100%|██████████████████████████████████████████████████████████████████████████| 300/300 [00:57<00:00,  5.21it/s] 
2023-09-10 19:38:38,427 - INFO: Extracting features.
100%|████████████████████████████████████████████████████████████████████████████| 94/94 [00:29<00:00,  3.17it/s] 
2023-09-10 19:39:08,444 - INFO: Evaluating 10000 episodes.
100%|████████████████████████████████████████████████████████████████████████████| 20/20 [00:22<00:00,  1.10s/it] 
2023-09-10 19:39:30,504 - INFO: Evaluation finished. 0.0022 seconds/episode
wandb: Waiting for W&B process to finish... (success).
wandb: 
wandb: Run history:
wandb: loss.NoHub █▁█▁█▁█▁█▁█▁█▁█▁█▁█▁█▁█▁█▁▃▁▃▁▃▁▃▁▃▁▃▁▃▁
wandb:
wandb: Run summary:
wandb:           accuracy.conf 0.00295
wandb:            accuracy.max 1.0
wandb:           accuracy.mean 0.69776
wandb:            accuracy.min 0.05333
wandb:            accuracy.std 0.1504
wandb: antihub_occurrence.conf 0.00025
wandb:  antihub_occurrence.max 0.1
wandb: antihub_occurrence.mean 0.01301
wandb:  antihub_occurrence.min 0.0
wandb:  antihub_occurrence.std 0.01288
wandb:            episode_time 0.00221
wandb:     hub_occurrence.conf 0.00093
wandb:      hub_occurrence.max 0.325
wandb:     hub_occurrence.mean 0.07859
wandb:      hub_occurrence.min 0.0
wandb:      hub_occurrence.std 0.04767
wandb:              loss.NoHub 6.99204
wandb:    robinhood_index.conf 0.00043
wandb:     robinhood_index.max 0.285
wandb:    robinhood_index.mean 0.18927
wandb:     robinhood_index.min 0.1175
wandb:     robinhood_index.std 0.02184
wandb:           skewness.conf 0.00476
wandb:            skewness.max 1.41923
wandb:           skewness.mean 0.29849
wandb:            skewness.min -0.60982
wandb:            skewness.std 0.24266
```
